src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -233,14 +240,15 @@ public bool ForEach(KeyValuePair<string, object> item)
                 switch (item.Key)
                 {
                     case SpanAttributeConstants.StatusCodeKey:
-                        this.StatusCode = item.Value as string;
+                        this.StatusCode = (StatusCode)item.Value;
+                        this.IsValid = Enum.IsDefined(typeof(StatusCode), item.Value);
```

Comment:
If you have time, we might want to take a look at what Enum.IsDefined does to the perf.

Reply:
Just pushed one change moving from the Enum.IsDefined to HashSet check. Below the benchmark:
|                       Method |      Mean |    Error |   StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------------------------- |----------:|---------:|---------:|-------:|------:|------:|----------:|
|                 HashSetCheck |  30.37 ns | 0.620 ns | 0.738 ns | 0.0057 |     - |     - |      24 B |
|     HashSetCheckDoesNotExist |  28.41 ns | 0.279 ns | 0.247 ns | 0.0057 |     - |     - |      24 B |
|             EnumDefinedCheck | 136.45 ns | 2.232 ns | 2.088 ns | 0.0057 |     - |     - |      24 B |
| EnumDefinedCheckDoesNotExist | 133.12 ns | 2.558 ns | 2.392 ns | 0.0057 |     - |     - |      24 B |

Reply:
HashSet has known perf issue on some version of the .NET Framework https://github.com/open-telemetry/opentelemetry-dotnet/issues/708#issuecomment-636540689.

Given the status code only has 3 values, I guess a special case optimization would be faster?

Reply:
just updated to a condition! :)